api/controller.py

```diff
@@ -667,19 +667,14 @@ def public_key_document(self):
 
 class OPDSFeedController(CirculationManagerController):
 
-    def groups(self, lane_identifier):
+    def groups(self, lane_identifier, feed_class=AcquisitionFeed):
```

Comment:
The methods in this class were streamlined a little bit and given the ability to mock the feed class whose methods we call.

We had really complicated end-to-end tests that stopped working when I merged in the core branch -- all these `AcquisitionFeed` methods used to end up going against the materialized view, and now they go against the search index. So without creating a real search index, we can't do an end-to-end test that verifies that e.g. the French book doesn't show up in the "Adult English Fiction" lane.

Fortunately, all that stuff is tested lower down, ultimately in `core/tests/test_external_search.py`, where we test that a `Filter` constructed a certain way will match certain books; and in `core/tests/test_lane.py`, where we test that a certain faceting object will modify a `Filter` in a certain way. So I changed the tests of these methods to mock the feed class and verify that the appropriate faceting objects (among other things) are passed in.

Overall I think this is a great change, but it does mean we don't have a full end-to-end test where we get an OPDS feed of all the books in a lane.